frontend/pages/ApiOnlyUser/ApiOnlyUser.tsx

```diff
@@ -1,21 +1,33 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { useDispatch } from "react-redux";
 import { push } from "react-router-redux";
-
+// @ts-ignore
+import { fetchCurrentUser, logoutUser } from "redux/nodes/auth/actions";
 import Button from "components/buttons/Button";
 import paths from "router/paths";
+import { IUser } from "interfaces/user";
 // @ts-ignore
 import fleetLogoText from "../../../assets/images/fleet-logo-text-white.svg";
 
 const baseClass = "api-only-user";
 
-const ApiOnlyUser = (): JSX.Element | null => {
+const ApiOnlyUser = (): JSX.Element => {
   const dispatch = useDispatch();
-  const { LOGIN } = paths;
-  const handleClick = (event: any) => dispatch(push(LOGIN));
+  const { LOGIN, HOME } = paths;
+  const handleClick = (event: any) => dispatch(logoutUser());
+
+  useEffect(() => {
+    dispatch(fetchCurrentUser()).then((user: any) => {
+      if (!user) {
+        dispatch(push(LOGIN));
+      } else if (user && !user.payload.user.api_only) {
```

Comment:
I think the naming could help ease understanding
```suggestion
    dispatch(fetchCurrentUser()).then((userResponse: any) => {
      if (!userResponse) {
        dispatch(push(LOGIN));
      } else if (userResponse && !userResponse.payload.user.api_only) {
```